Error compiling editor.cpp

With gcc810 in windows (MinGW), editor.cpp  goes wrong:

[ 97%] Building CXX object CMakeFiles/VVVVVV.dir/src/editor.cpp.obj
c:/VVVVVV-master/desktop_version/src/editor.cpp: In function
 'std::__cxx11::string find_tag(const string&, const string&, const string&)':
c:/VVVVVV-master/desktop_version/src/editor.cpp:180:39
: error: expected ')' before 'SCNx32'
             sscanf(number.c_str(), "%" SCNx32, &character);
                   ~                   ^~~~~~~
                                       )
c:/VVVVVV-master/desktop_version/src/editor.cpp:184:39
: error: expected ')' before 'SCNu32'
             sscanf(number.c_str(), "%" SCNu32, &character);
                   ~                   ^~~~~~~
                                       )
make[2]: *** [CMakeFiles/VVVVVV.dir/src/editor.cpp.obj] Error 1
make[1]: *** [CMakeFiles/VVVVVV.dir/all] Error 2
make: *** [all] Error 2

--- a/desktop_version/src/editor.cpp
+++ b/desktop_version/src/editor.cpp
@@ -23,6 +23,9 @@
 #ifndef __STDC_FORMAT_MACROS
 #define __STDC_FORMAT_MACROS
 #endif
+#ifndef _POSIX_SOURCE
+#define _POSIX_SOURCE
+#endif
 #include <inttypes.h>
 #include <cstdio>
 

--- a/desktop_version/src/editor.cpp
+++ b/desktop_version/src/editor.cpp
@@ -20,6 +20,9 @@
 #include <utf8/unchecked.h>
 #include <physfs.h>
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <inttypes.h>
 #include <cstdio>
 

